validator/client/validation_client.go

```diff
@@ -67,16 +67,17 @@ func (c *ValidationClient) Start(ctx context.Context) error {
 		return errors.New("couldn't read name from server")
 	}
 	var stylusArchs []string
-	if err := c.client.CallContext(ctx, &stylusArchs, server_api.Namespace+"_stylusArchs"); err != nil {
-		return err
-	}
-	if len(stylusArchs) == 0 {
-		return fmt.Errorf("could not read stylus archs from validation server")
-	}
-	for _, stylusArch := range stylusArchs {
-		if stylusArch != "wavm" && stylusArch != runtime.GOARCH && stylusArch != "mock" {
-			return fmt.Errorf("unsupported stylus architecture: %v", stylusArch)
+	if err := c.client.CallContext(ctx, &stylusArchs, server_api.Namespace+"_stylusArchs"); err == nil {
```

Comment:
I'd favor checking the error message here to prevent network errors from being caught here. In fact, I think we can probably downcast this into a type with the JSON-RPC error code, and there's a specific error code for a method not existing.